File: code/search/linear_search/linear_search.swift
//
//  linear_search.swift
//  test
//
//  Created by Kajornsak Peerapathananont on 10/14/2560 BE.
//  Copyright © 2560 Kajornsak Peerapathananont. All rights reserved.
//  Part of Cosmos by OpenGenus Foundation

import Foundation

/*
 Input : array of item and the element that want to be search
 Output : index of element (if found) or else -1
 
 
 */


func linear_search<T : Equatable>(array : [T], element : T) -> Int{
    for (index, item) in array.enumerated() {
        if(item == element){
            return index
        }
    }
    return -1
}

File: code/search/linear_search/linear_search.rs
fn search(vec: &[i32], n: i32) -> i32 {
    for i in 0..vec.len() {
        if vec[i] == n {
            return i as i32; // Element found return index
        }
    }
    return -1; // If element not found return -1
}

fn main() {
    let arr = vec![1, 3, 45, 56, 8, 21, 7, 69, 12];
    let num = 56;
    let index =  search(&arr, num);
    if index != -1 {
        println!("Found {}, at index {}", num, index);
    }else {
        println!("{} not found!", num)
    }
}